lib/srv/discovery/kube_integration_watcher.go

```diff
@@ -50,7 +51,14 @@ func (s *Server) startKubeIntegrationWatchers() error {
 
 	clt := s.AccessPoint
 
-	releaseChannels := automaticupgrades.Channels{}
+	pingResponse, err := s.AccessPoint.Ping(s.ctx)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	proxyPublicAddr := pingResponse.GetProxyPublicAddr()
+
+	releaseChannels := automaticupgrades.Channels{automaticupgrades.DefaultChannelName: &automaticupgrades.Channel{
+		ForwardURL: fmt.Sprintf("https://%s/webapi/automaticupgrades/channel/%s", proxyPublicAddr, automaticupgrades.DefaultCloudChannelName)}}
```

Comment:
I think we should use the "default channel" instead of the "cloud default channel" one. Default is mapped to cloudDefault in cloud, and returns the proxy version on self-hosted. This would make this code compatible with any kind of hosting.

Reply:
Updated https://github.com/gravitational/teleport/pull/43370/commits/01615796b2827cb9506d3de04335e4dd4837bd82